ipinfo/handler.py

```diff
@@ -120,9 +120,12 @@ def getDetails(self, ip_address=None, timeout=None):
             url += "/" + ip_address
         headers = handler_utils.get_headers(self.access_token, self.headers)
         response = requests.get(url, headers=headers, **req_opts)
-        if response.status_code == 429:
-            raise RequestQuotaExceededError()
-        if response.status_code >= 400:
+        if response.status_code >= 500:
```

Comment:
The only issue with this approach is that we lose the error details when `response.status_code >= 500`.

A better approach is to explicitly handle the response content type and extract the error message.

Could you change the lines (and the same in `handler_async.py`) to:
```
if response.status_code == 429:
    raise RequestQuotaExceededError()
if response.status_code >= 400:
    content_type = response.headers.get('Content-Type')
    error_code = response.status_code
    if content_type == 'application/json':
        error_response = response.json()
    else:
        error_response = {'error': response.text}
    raise APIError(error_code, json.dumps(error_response))
```

Thanks!

Reply:
Sure, good point. I'll update.